### PR TITLE
retry on conflict error in e2e test

### PR DESCRIPTION
**Pull request by cofyc**
### What problem does this PR solve? 

https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/operator_ghpr_e2e_test_kind/detail/operator_ghpr_e2e_test_kind/1987/tests/

```
/home/jenkins/agent/workspace/operator_ghpr_e2e_test_kind/go/src/github.com/pingcap/tidb-operator/tests/e2e/tidbcluster/tidbcluster.go:286
Jan 3 03:40:21.893: Expected update TiDB cluster
Unexpected error:
<*errors.StatusError | 0xc000358500>: {
ErrStatus: {
TypeMeta: {Kind: "", APIVersion: ""},
ListMeta: {
SelfLink: "",
ResourceVersion: "",
Continue: "",
RemainingItemCount: nil,
},
Status: "Failure",
Message: "Operation cannot be fulfilled on tidbclusters.pingcap.com \"service-it\": the object has been modified; please apply your changes to the latest version and try again",
Reason: "Conflict",
Details: {Name: "service-it", Group: "pingcap.com", Kind: "tidbclusters", UID: "", Causes: nil, RetryAfterSeconds: 0},
Code: 409,
},
}
Operation cannot be fulfilled on tidbclusters.pingcap.com "service-it": the object has been modified; please apply your changes to the latest version and try again
occurred
/home/jenkins/agent/workspace/operator_ghpr_e2e_test_kind/go/src/github.com/pingcap/tidb-operator/tests/e2e/tidbcluster/tidbcluster.go:308
```

### What is changed and how does it work?

retry on conflict error in e2e test

### Check List 

Tests 

 - Unit test
 - E2E test

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 1]
```diff
@@ -303,8 +304,15 @@ var _ = ginkgo.Describe("[tidb-operator] TiDBCluster", func() {
 		}
 
 		ginkgo.By(fmt.Sprintf("Adopt orphaned service created by helm"))
-		tc.Spec.TiDB.Service = &v1alpha1.TiDBServiceSpec{}
-		_, err = cli.PingcapV1alpha1().TidbClusters(ns).Update(tc)
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			tc, err := cli.PingcapV1alpha1().TidbClusters(ns).Get(tcName, metav1.GetOptions{})
```
Comment (aylei) [inline tests/e2e/tidbcluster/tidbcluster.go]: IIRC, the local variable `tc` will shadow the `tc` in the outer scope, the outer `tc` still remains un-updated and differs the actual `tc` spec in the cluster, which may  cause later mistake

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 2]
```diff
@@ -303,8 +304,15 @@ var _ = ginkgo.Describe("[tidb-operator] TiDBCluster", func() {
 		}
 
 		ginkgo.By(fmt.Sprintf("Adopt orphaned service created by helm"))
-		tc.Spec.TiDB.Service = &v1alpha1.TiDBServiceSpec{}
-		_, err = cli.PingcapV1alpha1().TidbClusters(ns).Update(tc)
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
```
Comment (aylei) [inline tests/e2e/tidbcluster/tidbcluster.go]: there is a utility method for this: 
```
err = controller.GuaranteedUpdate(cli, tc, func() error {
    tc.Spec.TiDB.Service = &v1alpha1.TiDBServiceSpec{}
    return nil
})
```

Comment (cofyc) [reply tests/e2e/tidbcluster/tidbcluster.go]: done, it's useful

**Pull request merged**